spec/controllers/admin/daily_menus_controller_spec.rb

```diff
@@ -0,0 +1,265 @@
+  end
+
+  describe 'authorized user' do
+    let(:food_items_attributes) { attributes_for_list(:food_item, 3) }
+    let(:daily_menu) { create(:daily_menu, food_items_attributes: food_items_attributes) }
+    let(:user) { create(:user) }
+
+    before do
+      user.update(is_admin: false)
+      sign_in(user)
+    end
+
+    describe 'GET /index' do
+      it 'redirects to root path with alert flash' do
+        get :index
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+
+    describe 'GET /show' do
+      it 'redirects to root path with alert flash' do
+        get :show, params: { id: daily_menu }
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+
+    describe 'GET /new' do
+      it 'redirects to root path with alert flash' do
+        get :new
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+
+    describe 'POST /create' do
+      it 'redirects to root path with alert flash' do
+        post :create, params: { id: daily_menu }
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+
+    describe 'GET /edit' do
+      it 'redirects to root path with alert flash' do
+        get :edit, params: { id: daily_menu }
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+
+    describe 'PATCH /update' do
+      it 'redirects to root path with alert flash' do
+        patch :update, params: { id: daily_menu }
+        expect(response).to redirect_to(root_path)
+        expect(flash[:alert]).to be_present
+      end
+    end
+  end
+
+  describe 'authorized admin' do
+    let(:food_items_attributes) { attributes_for_list(:food_item, 3) }
+    let(:daily_menu) { create(:daily_menu, food_items_attributes: food_items_attributes) }
+    let(:user) { create(:user) }
+
+    before do
+      user.update(is_admin: true)
+      sign_in(user)
+    end
+
+    describe 'GET /index' do
+      it 'renders index page' do
+        get :index
+        expect(response).to render_template(:index)
+      end
+    end
+
+    describe 'GET /show' do
+      it 'renders show page' do
+        get :show, params: { id: daily_menu }
+        expect(response).to render_template(:show)
+      end
+
+      it 'assigns requested daily_menu to @daily_menu' do
+        get :show, params: { id: daily_menu }
+        expect(assigns(:daily_menu)).to eq(daily_menu)
+      end
+    end
+
+    describe 'GET /new' do
+      it 'renders new page' do
+        get :new
+        expect(response).to render_template(:new)
+      end
+
+      it 'assigns new daily_menu to @daily_menu' do
+        get :new
+        expect(assigns(:daily_menu)).to be_a_new(DailyMenu)
+      end
+    end
+
+    describe 'POST /create' do
+      context 'valid food_items data' do
```

Comment:
context description usually starts with `with` or `when`
so has to be like `with valid food_items data`